storages/utils.py

```diff
@@ -117,3 +117,30 @@ def get_available_overwrite_name(name, max_length):
             'allows sufficient "max_length".' % name
         )
     return os.path.join(dir_name, "{}{}".format(file_root, file_ext))
+
+
+class ReadBytesWrapper:
+    """
+    A wrapper for a file-like object, that makes read() always returns bytes.
+    """
+    def __init__(self, file, encoding="utf-8"):
```

Comment:
Currently any string files will be encoded to utf-8 when saved.

I'm not sure about this. It could try to look up the `file.encoding` and use that.

Reply:
I think that's a good idea given how the `encoding` property is documented here https://docs.python.org/3/library/io.html#io.TextIOBase.encoding